src/h_vialib/secure/cookie.py

```diff
@@ -0,0 +1,122 @@
+"""Cookie helpers to create secure cookies."""
+
+from http.cookies import SimpleCookie
+
+from h_vialib.exceptions import MissingToken
+from h_vialib.secure.expiry import as_expires
+
+
+class Cookie:
+    """A simplified Cookie for security purposes."""
+
+    DATE_FORMAT = "%a, %d %b %Y %H:%M:%S %Z"
+
+    def __init__(self, name):
+        """Initialise this cookie object.
+
+        :param name: Cookie name to read and write
+        """
+        self._name = name
+
+    def create(self, value, expires=None, max_age=None):
+        """Create a cookie with a specified value and expiry.
+
+        This cookie is for security purposes and has a number of default
+        behaviors:
+
+         * The path is '/'
+         * HTTP Only is enabled
+         * The name is controlled to be the same as this class was intialised with
+
+        :param value: The value to store in the cookie
+        :param expires: Expiry time for the cookie
+        :param max_age: ... or max age of the cookie
+        :return: A tuple of the correct header and value to set this cookie
+
+        :raises ValueError: If no expiry is provided
+        """
+        cookie = SimpleCookie()
+
+        cookie[self._name] = value
+        cookie[self._name]["path"] = "/"
+        cookie[self._name]["httponly"] = True
+
+        if expires is not None:
+            if expires.tzinfo is None:
+                raise ValueError("The expires date must have a timezone")
+
+            cookie[self._name]["expires"] = expires.strftime(self.DATE_FORMAT)
+        elif max_age is not None:
+            cookie[self._name]["max-age"] = int(max_age)
+        else:
+            raise ValueError("You must specify an expiry time")
+
+        return tuple(cookie.output().split(": "))
+
+    def verify(self, cookies_header):
+        """Check for the presence of our cookie and get the value.
+
+        :param cookies_header: Full content of the "Cookies" header
+        :returns: The value matching our name, or None
+        """
+        if not cookies_header:
+            return None
+
+        cookie = SimpleCookie(cookies_header)
+        try:
+            return cookie[self._name].value
+        except KeyError:
+            return None
+
+
+class TokenBasedCookie(Cookie):
+    """A cookie which stores it's contents using a SecureToken."""
```

Comment:
You can stuff any `SecureToken` class in here and it will generate the guts of it. For usage example see: https://github.com/hypothesis/viahtml/pull/47/files#diff-7f58e17e4aea9a5e7e3bcc75c66b251e206df08202f7b66edb15c99cb1bc39e7R18